Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Parse/Blocks/CodeBlock.cs

```diff
@@ -0,0 +1,144 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Text;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Helpers;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Parse.Elements
+{
+    /// <summary>
+    /// Represents a block of text that is displayed in a fixed-width font.  Inline elements and
+    /// escape sequences are ignored inside the code block.
+    /// </summary>
+    public class CodeBlock : MarkdownBlock
```

Comment:
There's a lot of classes here that seems to be internal infrastructure. How many of them can we make `internal`